azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -565,6 +693,59 @@ def publish(self, topic, payload, qos=1, callback=None):
                 raise _create_error_from_rc_code(rc)
         self._op_manager.establish_operation(OperationType.PUBLISH, mid, callback)
 
+    def get_debug_status(self):
+        """
+        Return an infomrational status object that describes the current state of the Paho transport
+        """
+
+        self._paho_status.time_since_last_paho_traffic_in = format_time_delta(
+            self._mqtt_client._last_msg_in
+        )
+        self._paho_status.time_since_last_paho_traffic_out = format_time_delta(
+            self._mqtt_client._last_msg_out
+        )
+
+        self._paho_status.client_object_id = id(self._mqtt_client)
+        self._paho_status.thread_name = (
+            self._mqtt_client._thread.name if self._mqtt_client._thread else "None"
+        )
+        self._paho_status.thread_is_alive = (
+            str(self._mqtt_client._thread.is_alive()) if self._mqtt_client._thread else "No thread"
+        )
+        self._paho_status.len_out_mesage_queue = len(self._mqtt_client._out_messages)
+        self._paho_status.len_in_message_queue = len(self._mqtt_client._in_messages)
+        self._paho_status.len_out_pakcet_queue = len(self._mqtt_client._out_packet)
+        self._paho_status.thread_terminate = self._mqtt_client._thread_terminate
+        self._paho_status.paho_connection_state = self._mqtt_client._state
+
+        return self._paho_status
+
+    def get_debug_config(self):
+        """
+        Return an infomrational status object that describes the configuration of the Paho transport
+        """
+        config = PahoConfig()
+
+        config.transport = self._mqtt_client._transport
```

Comment:
I feel as though accessing convention-private attributes of Paho is dangerous. There is no guarantee they don't change in an update to Paho, which could break anyone using our library until we make an update. Furthermore, that risk doesn't really seem worth it for a logging utility.

Reply:
I agree that it's dangerous. I also think it's super useful -- at least some of these are. Using getattr() with a default would be much safer.